### DISCUSSION TITLE
Temperature different between Modbus and app

### DISCUSSION THREAD

**Discussion by Carl-mgb66** [nathanmarlor/foxess_modbus#710] [Q&A]
Has anyone else been experiencing significant differences between the Modbus (and cloud) temperatures and those reported by the Fox app?

It's something I'm seeing and others also seem to be: https://community.home-assistant.io/t/foxess-modbus-and-foxcloud-different-temps/782923?u=mgb66

Currently Modbus is reporting a battery temperature of 17 degC, whereas the app says 6.9 degC. The  lower temperature seems more plausible as I have a separate sensor in the same space measuring 8 degC.

**Comment by FozzieUK** [comment]
Which inverter do you have ?

The app is showing the lowest cell temperature in the pack and it sounds like you are looking at the battery_temp which despite it's name is actually the temperature measured at the BMS (and usually measures 6C-10C higher) - depending on the inverter version you have the low cell temp may (or may not be supported) - if it is supported you will find it in the sensors as `sensor.bms_cell_temp_low`

**Reply by Carl-mgb66** [reply]
My inverter is a H1-5.0-E and the battery is a ECS2900-2. I don't appear to have the low cell temperature sensor, but the cloud (web page) and Modbus are showing:

- Ambient Temperature: 26.1 degC
- Inverter Temperature: 16.2 degC
- Battery Temperature: 17.3 degC

Whereas the App is showing a battery temperature of 7 degC.

The battery is at 14% and not charging or exporting.

My other separate temperature sensors in the same place as the battery/inverter are reading 9.5 degC (i.e. the ambient temperature in the space).

I could have sworn they used to be closer.

**Reply by FozzieUK** [reply]
Yes you’re right they used to be the same,  Fox switched to using the lowest cell temp about 6 months ago in the V2 app, before that it used to show Battery Temperature (as reported by the BMS).

Is you inverter the H1 G2 version ?  (approx less than 12 months old) - that inverter doesn’t yet support the low and high battery temps (or the low and high battery cell voltages).

But if you have the G1 inverter, in HA try going to Settings, Devices & Services and click on ‘1 DEVICE’ for the Foxess Modbus integration. At the bottom of the page it will say ‘+xx entities not shown’ click that and if it shows BMS Cell Temp Low click on that and slide the ‘enabled’ switch on and then update - if it is capable of reading it you should find after 30 seconds you have another entity `BMS Cell Temp Low`, entity_id = `sensor.bms_cell_temp_low`

**Reply by Carl-mgb66** [reply]
Mine is older than 12 months.

I don't have that sensor, maybe because I connect my Modbus via Direct LAN?

I can understand the battery discrepancy, the ambient one seems very odd.

**Reply by FozzieUK** [reply]
Ok, yes if you have LAN you only get a limited set of sensors.

The confusion around ambient is the name, Fox have named it 'ambient temperature' but it is actually the temperature measured by the internal probe of the inverter just near the CPU so effectively it is the inner temperature of the inverter (which will be reasonably warm all year). 
The other temperature sensor you get is called Inverter Temperature and is actually the temperature of the cooling fans on the rear of the inverter - you'll see that go up dramatically when it is working hard.

**Reply by Carl-mgb66** [reply]
Ok, thanks, makes more sense now.
